cmd/selfsync_test.go

```diff
@@ -0,0 +1,31 @@
+package cmd
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestFetchChannels(t *testing.T) {
+	res, err := fetchChannels("620280", StatusFailed)
+	if err != nil {
+		t.Error(err)
+	}
+	if res == nil {
+		t.Error("empty response")
+	}
+	fmt.Println(res)
+}
+
+// warning this test will actually set sync_server on the db entry for this test channel (mine)
+// such field should be reset to null if the test must be run on a different machine (different hostname)
+// and obviously the auth token must be appropriate
+func TestSetChannelSyncStatus(t *testing.T) {
+	err := setChannelSyncStatus("620280", "UCNQfQvFMPnInwsU_iGYArJQ", StatusSyncing)
```

Comment:
are you making calls to the live API here? this is a very weak test. and you're including the auth_token in public code that's visible on github?

Reply:
just to make this clear. the first parameter is the auth token, which is just a number as I'm using this "test" locally. The second parameter is a channel ID.

I basically wrote tests to run the portion of code without having to compile and run the whole program, I believe more code should be tested in a better way. I don't think any of the tests here actually test anything.

Also yesh, I'm calling a local API, not the live API, but the right approach would be to mock the API.